src/context/actions.js

```diff
@@ -0,0 +1,260 @@
+  }
+}
+
+const storeComicData = async value => {
+  try {
+    const jsonValue = JSON.stringify(value);
+    await AsyncStorage.setItem('@FavoritedComics', jsonValue);
+  } catch (error) {
+    setErrorData(error);
+  }
+};
+export async function getComicData() {
+  const newArr=[]
+  try {
+    let jsonValue = await AsyncStorage.getItem('@FavoritedComics');
+    return jsonValue === null ? newArr : JSON.parse(jsonValue);
+  } catch (error) {
+    setErrorData(error);
+  }
+}
+
+export async function getLanguageFromLocalStorage() {
+  try {
+    let languageData = await AsyncStorage.getItem('@Language');
+    return languageData;
+  } catch (error) {
+    setErrorData(error);
+  }
+}
+
+export async function setLanguageToLocalStorage(value) {
+  try {
+    await AsyncStorage.setItem('@Language', value);
+  } catch (error) {
+    setErrorData(error);
+  }
+}
+
+export async function getModeFromLocalStorage() {
+  try {
+    let modeData = await AsyncStorage.getItem('@Mode');
+    return modeData;
+  } catch (error) {
+    setErrorData(error);
+  }
+}
+
+export async function setModeToLocalStorage(value) {
+  try {
+    await AsyncStorage.setItem('@Mode', value);
+  } catch (error) {
+    setErrorData(error);
+  }
+}
+
+// ----------------------------------------------------------------------------
+
+export const setLoading = () => ({
+  type: 'SET_LOADING',
+});
+
+export const setHeroData = heroList => ({
+  type: 'SET_HEROES',
+  payload: heroList,
+});
+export const setComicData = comicList => ({
+  type: 'SET_COMICS',
+  payload: comicList,
+});
+
+export const setFavoriteHeroesToState = favoriteHeroesList => ({
+  type: 'SET_FAVORITE_HEROES',
+  payload: favoriteHeroesList,
+});
+
+export const setFavoriteComicsToState = favoriteComicsList => ({
+  type: 'SET_FAVORITE_COMICS',
+  payload: favoriteComicsList,
+});
+export const setLanguageToState = languageData => ({
+  type: 'SET_LANGUAGE',
+  payload: languageData,
+});
+export const setModeToState = modeData => ({
+  type: 'SET_MODE',
+  payload: modeData,
+});
+
+// export const getHeroesFromApi = (heroData, dispatch) => {
+//   const {heroes, heroError, heroLoading} = heroData;
+
+//   if (heroLoading) {
+//     dispatch(setLoading());
+//   } else {
+//     if (!!heroError) {
+//       dispatch(setErrorData(heroError));
+//     }
+//     dispatch(setHeroData(heroes));
+//   }
+// };
+
+// export const getComicsFromApi = (comicData, dispatch) => {
+//   const {comics, comicError, comicLoading} = comicData;
+
+//   if (comicLoading) {
+//     dispatch(setLoading());
+//   } else {
+//     if (!!comicError) {
+//       dispatch(setErrorData(comicError));
+//     }
+//     dispatch(setComicData(comics));
+//   }
+// };
+
+export const setFavoriteHeroList = (hero, dispatch) => {
+  dispatch(setLoading());
+  let favoriteHeroList = [];
+  getHeroData().then(data => {
+    favoriteHeroList = data;
+
+    if (favoriteHeroList.length === 0) {
+      let updatedList = [hero];
```

Comment:
Değişmeyen datalar için let kullanmak sağlıklı değil demiştik, aşağıda da bolca let kullanılmış hep.